### PR TITLE
CatalogDetail Tests

### PR DESCRIPTION
**Pull request by tmerlier**
Add tests to CatalogDetail component

### CODE REVIEW

**Comment by tmerlier** [comment]
@jdesboeufs

File: src/CatalogDetail/CatalogDetail.js [hunk 1]
```diff
@@ -5,39 +5,25 @@ import CatalogSection from '../Section/CatalogSection/CatalogSection'
 import StatisticsSection from '../Section/StatisticsSection/StatisticsSection'
 import OrganizationsSection from '../Section/OrganizationsSection/OrganizationsSection'
 import HarvestsSection from '../Section/HarvestsSection/HarvestsSection'
+import fetchCatalog from '../fetch/fetchCatalog'
+import fetchMetrics from '../fetch/fetchMetrics'
 
 class CatalogDetail extends Component {
   constructor(props) {
     super(props)
     this.state = {catalog: undefined, metrics: undefined}
```
Comment (jdesboeufs) [inline src/CatalogDetail/CatalogDetail.js]: Bad practice.
Consider using `getInitialState()`.
Moreover setting values to `undefined` can be easily work-arounded.

File: src/CatalogDetail/CatalogDetail.js [hunk 2]
```diff
@@ -5,39 +5,25 @@ import CatalogSection from '../Section/CatalogSection/CatalogSection'
 import StatisticsSection from '../Section/StatisticsSection/StatisticsSection'
 import OrganizationsSection from '../Section/OrganizationsSection/OrganizationsSection'
 import HarvestsSection from '../Section/HarvestsSection/HarvestsSection'
+import fetchCatalog from '../fetch/fetchCatalog'
```
Comment (jdesboeufs) [inline src/CatalogDetail/CatalogDetail.js:8]: In the future, those functions could be grouped in a module.

File: src/CatalogDetail/CatalogDetail.js [hunk 3]
```diff
@@ -5,39 +5,25 @@ import CatalogSection from '../Section/CatalogSection/CatalogSection'
...
   }
 
-  getCatalog() {
-    if (!this.state.catalog) {
-    return fetch(`https://inspire.data.gouv.fr/api/geogw/catalogs/${this.props.params.catalogId}`)
-      .then((response) => response.json())
-      .then((catalog) => {
-        this.setState({catalog})
+  componentWillMount() {
+    return fetchCatalog(this.props.params.catalogId)
```
Comment (jdesboeufs) [inline src/CatalogDetail/CatalogDetail.js:38]: The two `fetch` can be processed in parallel. Consider using `Promise.all`.
You should break this into two component methods: one for Catalog, one for Metrics.

File: src/CatalogDetail/CatalogDetail.js [hunk 4]
```diff
@@ -5,39 +5,25 @@ import CatalogSection from '../Section/CatalogSection/CatalogSection'
...
-        })
-        .catch((err) => {
-          console.error(err)
-        })
-      }
+      .then(fetchMetrics(this.props.params.catalogId)
+        .then(metrics => {
+          this.setState({ metrics })
+        }))
+      .catch(err => console.error(err));
```
Comment (jdesboeufs) [inline src/CatalogDetail/CatalogDetail.js]: `console.error` is a bad practice since the final user has no access to console.

**Review by jdesboeufs — CHANGES_REQUESTED** [review]

**Comment by tmerlier** [comment]
@jdesboeufs

**Review by jdesboeufs — APPROVED** [review]

**Pull request merged**